routers/repo/view.go

```diff
@@ -210,9 +211,26 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 			}
 
 			var output bytes.Buffer
-			lines := strings.Split(fileContent, "\n")
+			var terminator string
```

Comment:
Actually, yes, this looks to work. My first approach was similar to this one, but for some reason I stopped pursuing it. Anyways, this copies fine, preserves newlines in mixed files and also highlights fine, looking good!